client/app/iplb/sslCertificate/order/iplb-ssl-certificate-order.controller.js

```diff
@@ -1,50 +1,102 @@
 
-        this.newSsl = {
-            type: this.sslTypes[0]
-        };
+    init () {
+        this.paidOffers.load();
     }
 
     order () {
         if (this.form.$invalid) {
             return this.$q.reject();
         }
-        this.saving = true;
 
-        let orderConfig;
-        if (this.newSsl.type === "ev") {
-            orderConfig = this.IpLoadBalancerConstant.sslOrders.comodoEv;
-        } else if (this.newSsl.type === "dv") {
-            orderConfig = this.IpLoadBalancerConstant.sslOrders.comodoDv;
-        } else if (this.newSsl.type === "free") {
-            orderConfig = this.IpLoadBalancerConstant.sslOrders.free;
+        this.planCode = this.newSsl.type.replace(/_/g, "-");
+
+        if (this.planCode === "free") {
+            return this.orderFreeCertificate();
+        }
+
+        const sslOffer = this.offers.find(offer => offer.planCode === this.planCode);
+
+        if (!sslOffer) {
+            return null;
         }
 
-        orderConfig.configuration = {
-            domainname: this.newSsl.displayName ? this.newSsl.displayName : this.newSsl.fqdn,
-            backendserver: this.newSsl.fqdn
-        };
+        const options = Object.assign(_.pick(sslOffer.prices[0], [
+            "duration",
+            "pricingMode"
+        ]), {
+            planCode: this.planCode,
+            quantity: 1
+        });
 
-        if (this.newSsl.type === "ev") {
-            orderConfig.option[0].configuration = this.newSsl.option;
-            orderConfig.option[0].configuration.commonName = this.newSsl.displayName;
-            orderConfig.option[0].configuration.dcv_email = this.newSsl.option.email;
-            orderConfig.option[0].configuration.country = this.newSsl.option.countryName;
+        return this.orderPaidCertificate(options);
+    }
+
+    orderFreeCertificate () {
+        const fqdn = this.newSsl.fqdn.split(",");
+
+        this.saving = true;
+        this.IpLoadBalancerSslCertificateService.orderFreeCertificate(this.$stateParams.serviceName, fqdn)
+            .then(() => this.$state.go("network.iplb.detail.ssl-certificate.home"))
+            .finally(() => {
+                this.saving = false;
+            });
+    }
+
+    orderPaidCertificate (options) {
+        const configuration = Object.assign({}, this.newSsl);
+
+        if (this.planCode === "iplb-ssl-ev-single") {
+            configuration.commonName = configuration.fqdn;
+            configuration.dcv_email = configuration.email;
+            configuration.country = configuration.countryName;
+            delete configuration.fqdn;
         }
 
-        return this.OrderHelperService.openExpressOrderUrl(orderConfig);
+        delete configuration.type;
+
+        this.saving = true;
+        this.IpLoadBalancerSslCertificateService.orderPaidCertificate(this.$stateParams.serviceName, options, configuration)
+            .then(result => {
+                location.href = result.url;
+            })
+            .catch(() => {
```

Comment:
CF previous comment

Reply:
Here, we load a new page, and it take a few seconds!
So, instead of enable the submit button in finally (no matter if it's a success or an error), I just keep it disabled if there is no error, to prevent the user te re-click on it during new page loading.

Reply:
Thanks for the explaination :)